### PR TITLE
Adding TriggerAuth example for pubsub

### PR DESCRIPTION
**Pull request by harshthakur9030**
Signed-off-by: Harsh Thakur <harshthakur9030@gmail.com>

### CODE REVIEW

File: content/docs/2.0/scalers/gcp-pub-sub.md [hunk 1]
```diff
@@ -48,3 +54,36 @@ spec:
...
+  name: keda-trigger-auth-gcp-credentials
+spec:
+  secretTargetRef:
+  - parameter: GoogleApplicationCredentials 
+    name: pubsub-secret        # Required. Refers to the name of the secret
+    key: GOOGLE_APPLICATION_CREDENTIALS_JSON       # Required.
+
+
+---
+
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/gcp-pub-sub.md:73]:
```suggestion
---
```

File: content/docs/2.0/scalers/gcp-pub-sub.md [hunk 2]
```diff
@@ -48,3 +54,36 @@ spec:
...
+  scaleTargetRef:
+    name: keda-pubsub-go
+  triggers:
+  - type: gcp-pubsub
+    authenticationRef:
+      name: keda-trigger-auth-gcp-credentials
+    metadata:
+      subscriptionName: "input" # Required  
+
+
```
Comment (tomkerkhove) [inline content/docs/2.0/scalers/gcp-pub-sub.md:88]:
```suggestion
```

**Review by tomkerkhove — APPROVED** [review]
LGTM, just added 2 nits

**Comment by harshthakur9030** [comment]
Applied the suggestions. @tomkerkhove

**Comment by tomkerkhove** [comment]
We're good to go, thanks! Can you just fix DCO issues please?

**Comment by harshthakur9030** [comment]
Oops! That happened cause I just applied the suggestions from UI. Anyway, fixed it

**Pull request merged**